### ISSUE TITLE
Provider detection does not always work

### ISSUE THREAD

**Issue by mcanevet** [camptocamp/bivac#186]
Somehow sometimes the output of the command to detect the provider looks empty even if it is not.

**Comment by mcanevet** [comment]
I have the same issue with Cattle orchestrator:
```
5/24/2018 3:13:52 PMtime="2018-05-24T13:13:52Z" level=info msg="Detecting provider" volume=dockercomposerancherserver_mysql
5/24/2018 3:13:59 PMtime="2018-05-24T13:13:59Z" level=debug msg="01 2018-05-24T13:13:53.263114000Z mysql\n" cmd="sh -c ([[ -d /var/lib/mysql/mysql ]] && echo -n 'mysql') || ([[ -f /var/lib/mysql/PG_VERSION ]] && echo -n 'postgresql') || ([[ -f /var/lib/mysql/DB_CONFIG ]] && echo -n 'openldap'); return 0" container=1i1923797 volumes="dockercomposerancherserver_mysql:/var/lib/mysql"
5/24/2018 3:13:59 PMtime="2018-05-24T13:13:59Z" level=info msg="Found data provider" provider=Default volume=dockercomposerancherserver_mysql
5/24/2018 3:13:59 PMtime="2018-05-24T13:13:59Z" level=debug msg="Container found using volume" container=1i1695548 volume=dockercomposerancherserver_mysql
5/24/2018 3:13:59 PMtime="2018-05-24T13:13:59Z" level=info msg="No prepare command to execute in container" container=1i1695548 volume=dockercomposerancherserver_mysql
```

**Comment by cryptobioz** [comment]
The problem with Cattle should be fixed with the patch #198. I can't reproduce on Minishift or Minikube but the worker is running as root, could it be a rights issue?

**Comment by mcanevet** [comment]
Fixed by #198
